integration-tests/src/test/gate-token.test.ts

```diff
@@ -0,0 +1,108 @@
+import { KeyringPair, creditcoinApi, Keyring, BN, Balance } from 'creditcoin-js';
+import { AccountId, Blockchain } from 'creditcoin-js/lib/model';
+import { CreditcoinApi, VerificationError } from 'creditcoin-js/lib/types';
+import { checkAddress, testData, tryRegisterAddress } from 'creditcoin-js/lib/testUtils';
+
+import { extractFee, testIf } from '../utils';
+
+import { deployGATEToken } from '../ctc-deploy';
+import { JsonRpcProvider } from '@ethersproject/providers';
+import { Signer, Wallet, Contract } from 'ethers';
+import { mnemonicGenerate } from '@polkadot/util-crypto';
+import { personalSignAccountId, signAccountId } from 'creditcoin-js/lib/utils';
+import { ethSignSignature, personalSignSignature } from 'creditcoin-js/lib/extrinsics/register-address-v2';
+import { requestCollectCoins } from 'creditcoin-js/lib/extrinsics/request-collect-coins';
+import { GATEContract } from 'creditcoin-js/lib/extrinsics/request-collect-coins-v2';
+
+describe('Test GATE Token', (): void => {
+    let ccApi: CreditcoinApi;
+    let sudoSigner: KeyringPair;
+
+    // Needed to interact with the ethererum private node
+    const testingData = testData(
+        (global as any).CREDITCOIN_ETHEREUM_CHAIN as Blockchain,
+        (global as any).CREDITCOIN_CREATE_WALLET,
+    );
+    const { keyring } = testingData;
+    const provider = new JsonRpcProvider((global as any).CREDITCOIN_ETHEREUM_NODE_URL);
+    const deployer = new Wallet((global as any).CREDITCOIN_CTC_DEPLOYER_PRIVATE_KEY, provider);
+    const burnAmount = 200;
+
+    // Holds the reference to the deployed GATE contract
+    let gateToken: any;
+    const gateKeyring = new Keyring({ type: 'ed25519', ss58Format: 3 });
+    const gateFaucet = gateKeyring.addFromUri(mnemonicGenerate(12));
+
+    beforeAll(async () => {
+        gateToken = await deployGATEToken(deployer);
+
+        ccApi = await creditcoinApi((global as any).CREDITCOIN_API_URL);
+        if ((global as any).CREDITCOIN_EXECUTE_SETUP_AUTHORITY) {
+            sudoSigner = (global as any).CREDITCOIN_CREATE_SIGNER(keyring, 'lender');
+        }
+    });
+
+    afterAll(async () => {
+        await ccApi.api.disconnect();
+    });
+
+    test('End to end', async () => {
```

Comment:
I think this should be  `testIf((global as any).CREDITCOIN_EXECUTE_SETUP_AUTHORITY`

b/c we can't call sudo on mainnet, probably not on Testnet after the latest changes.

Reply:
Done